src/lib/atom-private.c

```diff
@@ -1025,6 +1029,33 @@ _lldpctl_atom_get_str_port(lldpctl_atom_t *atom, lldpctl_key_t key)
 	}
 }
 
+static lldpctl_atom_t*
+_lldpctl_atom_set_str_port(lldpctl_atom_t *atom, lldpctl_key_t key,
+    const char *value)
+{
+	struct _lldpctl_atom_port_t *p =
+	    (struct _lldpctl_atom_port_t *)atom;
+	struct lldpd_port     *port     = p->port;
+
+	if (!value || !strlen(value))
+		return NULL;
+
+	switch (key) {
+	case lldpctl_k_port_id:
+		if (port->p_id != NULL) {
+			free(port->p_id);
```

Comment:
Actually, it's the reason I added the NULL checks [since I took them as reference].
I wasn't sure why they were there; I assumed that maybe some other libc implementation of free() does not NULL check.
Anyway; will replace these with simple free() calls.